nilearn/input_data/__init__.py

```diff
@@ -1,11 +1,31 @@
 """
-The :mod:`nilearn.input_data` module includes scikit-learn tranformers.
+The :mod:`nilearn.input_data` module used to include masker objects.
+It is deprecated since release 0.9.0 in favor of the
+:mod:`~nilearn.maskers` module.
+
+Please consider updating your code:
+
+.. code-blocks::python
+    from nilearn.input_data import NiftiMasker
+
+becomes:
+
+.. code-blocks::python
+    from nilearn.maskers import NiftiMasker
+
+Note that all imports that used to work will continue to do so with
+a simple warning.
 """
-from .nifti_masker import NiftiMasker
-from .multi_nifti_masker import MultiNiftiMasker
-from .nifti_labels_masker import NiftiLabelsMasker
-from .nifti_maps_masker import NiftiMapsMasker
-from .nifti_spheres_masker import NiftiSpheresMasker
+import warnings
+
+
+from nilearn.maskers import (
```

Comment:
backwards compatibility must not be broken without a deprecation cycle for anything that is in the public API currently. For example WDYT of the `input_data` submodules, as their name does not start with an underscore?
```
from nilearn.input_data import nifti_maps_masker
```
from https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces:
> Even with `__all__` set appropriately, internal interfaces (packages, modules, classes, functions, attributes or other names) should still be prefixed with a single leading underscore.

I'm not sure everyone would consider these modules internal just because their name isn't in `__all__`

Reply:
Thanks @jeromedockes, this is a really good point.
I think the new implementation (see 253f20b) should solve these issues.